src/Traitor.jl

```diff
@@ -54,56 +54,68 @@ previous definition.
 """
 module Traitor
 
-import Base: @pure
-export @traitor, @betray, supertrait
+using Base: @pure, uncompressed_ast, unwrap_unionall, tuple_type_cons
+using Core: SimpleVector, svec, CodeInfo
 
+export @traitor, supertrait, betray!
 
 """
-    @betray f
+    betray!(f, ::Type{TT}) where {TT <: Tuple}
 
-The goal of the `@betray` macro is to take over the methods of `f` and ready
-them to be compatible with traits-based dispatch, so that new submethods
-defined by `@traitor` do not overwrite existing default methods (and used where
-no more specific trait match is found).
+Et tu?
 
-WARNING: Please note this macro has NOT been implemented yet! (If anyone knows
-how to turn a `Method` object into a new method definition, please let us know).
+Take over the methods of `f` and readythem to be compatible with traits-based dispatch, 
+so that new submethods defined by `@traitor` do not overwrite existing default methods 
+(and used where no more specific trait match is found).
 
-(See also `Traitor` and `@traitor`.)
+Betraying functions is not a very kind thing to do, and can cause unintended side effects.
 """
-macro betray(ex)
-    if !isa(ex, Symbol)
-        error("Use @betray functionname")
+function betray!(f, ::Type{TT}) where {TT<:Tuple}
+    whereparams = TT isa UnionAll ? typevars(TT.body) : tuple()
+    if !isempty(whereparams)
+        throw("Where parameters not currently supported")
     end
+    # TODO: make this work with TT of the form (Tuple{T, T} where T)
+    
+    m = (last ∘ collect ∘ methods)(f, TT)
```

Comment:
Why `last`? Should it be `only`?

Also, purely out of curiousity, do you find this easier to read than `last(collect(methods(f, TT)))`?

Reply:
I use `last` because `methods` will give all the more specific methods as well as the one you ask for. For example,
```julia
julia> methods(sin, Tuple{Real})
# 4 methods for generic function "sin":
[1] sin(a::Float16) in Base.Math at math.jl:1159
[2] sin(x::T) where T<:Union{Float32, Float64} in Base.Math at special/trig.jl:29
[3] sin(x::BigFloat) in Base.MPFR at mpfr.jl:728
[4] sin(x::Real) in Base.Math at math.jl:404
```
Here, we just wanted the last one. 

> Also, purely out of curiousity, do you find this easier to read than `last(collect(methods(f, TT)))`?

Slightly yeah. Even with coloured parens, I find it hard for my eye to quickly verify that they're all matched once you have that many nested calls. It's not a huge deal either way for me though.